warehouse/tuf/__init__.py

```diff
@@ -0,0 +1,103 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import time
+
+from http import HTTPStatus
+
+import requests
+
+from pyramid.request import Request
+
+from warehouse.packaging.models import Project
+from warehouse.packaging.utils import render_simple_detail
+
+
+def _rstuf_is_bootstrapped(server: str) -> bool:
+    """Call RSTUF bootstrap API to check, if RSTUF is ready to be used."""
+    response = requests.get(f"{server}/api/v1/bootstrap")
+    if response.status_code != HTTPStatus.OK:
+        return False
+
+    return response.json()["data"]["bootstrap"]
+
+
+def _rstuf_wait_for_success(server: str, task_id: str) -> None:
+    """Poll RSTUF status API until async task finishes.
+
+    Raises RuntimeError, if task fails.
+    """
+    while True:
```

Comment:
Is this in theory of state not being changed infinite loop?

Reply:
Yeah, this is just a stopgap solution. We should definitely do better. See discussion item about async task handling in PR description.